docs/current_docs/api/constructors.mdx

```diff
@@ -126,12 +117,26 @@ Here is an example of a Dagger module with a default constructor argument of typ
 ```python file=./snippets/constructors/python/default-object/main.py
 ```
 
+:::info
+You can use other [`@dataclasses.dataclass`](https://docs.python.org/3/library/dataclasses.html) decorators to generate more complex constructors. For example:
+
+```python file=./snippets/constructors/python/complex/main.py
+```
+
+A few important notes:
+
+- Class attributes become constructor arguments by default, unless specified as `field(init=False)`.
+- The [`__post_init__` function](https://docs.python.org/3/library/dataclasses.html#post-init-processing) is called at the end of the auto-generated constructor. So by setting `init=False` on the container argument, it changes from being a constructor argument to a regular argument that is still required.
```

Comment:
It's incorrect that it changes from being a constructor argument to a regular argument. It's both if `init=True`.
```suggestion
- Attributes are both fields and constructor arguments by default. So by setting `init=False` on the container argument, it's no longer added as a constructor argument but is still a required field that must be initialized if there's no default value.
- The [`__post_init__` function](https://docs.python.org/3/library/dataclasses.html#post-init-processing) is called at the end of the auto-generated constructor which can be used, for example, to initialize field values that depend on other field values.
```